api/src/test/java/com/foro/api/TestTopicRepositoryIT.java

```diff
@@ -0,0 +1,149 @@
+        Integer idUser = 3;
+        Course course = Course.SPRINGBOOT;
+        Boolean deleteTopic = false;
+
+        TopicDTO topicInformation = TopicDTO.builder()
+                .title(title)
+                .message(message)
+                .creationDate(creationDate)
+                .topicStatus(topicStatus)
+                .idUser(idUser)
+                .course(course)
+                .deleteTopic(deleteTopic)
+                .build();
+
+        TopicDTO topic = topicRepository.save(topicInformation);
+        TopicDTO topicInformationInitial=new TopicDTO(topicInformation.getId(), idUser,title,message,creationDate,topicStatus,course,deleteTopic);
+
+
+        String titleModified = "What means SpringBoot";
+        String messageModified = "I dont understand the difference between Spring and SpringBoot";
+        String creationDateStringModified = "2021/05/04 11:30:00";
+        DateTimeFormatter formatterModified = DateTimeFormatter.ofPattern("yyyy/MM/dd HH:mm:ss");
+        LocalDateTime creationDateModified = LocalDateTime.parse(creationDateStringModified, formatterModified);
+        TopicStatus topicStatusModified = TopicStatus.CLOSED;
+        Integer idUserModified = 3;
+        Course courseModified = Course.SPRINGBOOT;
+        Boolean deleteTopicModified = false;
+
+        TopicDTO topicToBeModified = topicRepository.getReferenceById(topicInformation.getId());
+
+        topicToBeModified.setTitle(titleModified);
+        topicToBeModified.setMessage(messageModified);
+        topicToBeModified.setCreationDate(creationDateModified);
+        topicToBeModified.setTopicStatus(topicStatusModified);
+        topicToBeModified.setIdUser(idUserModified);
+        topicToBeModified.setCourse(courseModified);
+        topicToBeModified.setDeleteTopic(deleteTopicModified);
+
+        topicRepository.save(topicToBeModified);
+        Assert.assertEquals(topicToBeModified.getTitle(),"What means SpringBoot");
+        assertEqualsTopic(topicInformationInitial, topicToBeModified);
+
+        topicRepository.deleteById(topicToBeModified.getId());
+        boolean topicDeleted=topicRepository.findById(topicToBeModified.getId()).isPresent();
+        Assert.assertFalse(topicDeleted);
+    }
+    public void assertEqualsTopic(TopicDTO topicToBeCompared,TopicDTO topicWhoComparesTo){
+        Assert.assertEquals(topicToBeCompared.getId(), topicWhoComparesTo.getId());
+        Assert.assertEquals(topicToBeCompared.getMessage(), topicWhoComparesTo.getMessage());
+        Assert.assertEquals(topicToBeCompared.getCreationDate(), topicWhoComparesTo.getCreationDate());
+        Assert.assertEquals(topicToBeCompared.getTopicStatus(), topicWhoComparesTo.getTopicStatus());
+        Assert.assertEquals(topicToBeCompared.getIdUser(), topicWhoComparesTo.getIdUser());
+        Assert.assertEquals(topicToBeCompared.getCourse(), topicWhoComparesTo.getCourse());
+        }
+        @Test
+    public void testListTopics(){
```

Comment:
Bad formatting on `@Test`, additionally why this test didn't need the `rollback` and `@Transactional`?...